python/cuml/naive_bayes/naive_bayes.py

```diff
@@ -0,0 +1,533 @@
+        self.feature_count_ = cp.zeros((n_effective_classes, n_features),
+                                       order="F", dtype=dtype)
+
+    def _count(self, X, Y):
+        """
+        Sum feature counts & class prior counts and add to current model.
+
+        Parameters
+        ----------
+        X : cupy.ndarray or cupy.sparse matrix of size
+                  (n_rows, n_features)
+        Y : cupy.array of monotonic class labels
+        """
+
+        if X.ndim != 2:
+            raise ValueError("Input samples should be a 2D array")
+
+        if Y.dtype != self.classes_.dtype:
+            warnings.warn("Y dtype does not match classes_ dtype. Y will be "
+                          "converted, which will increase memory consumption")
+
+        counts = cp.zeros((self.n_classes_, self.n_features_),
+                          order="F", dtype=X.dtype)
+
+        class_c = cp.zeros(self.n_classes_, order="F", dtype=X.dtype)
+
+        n_rows = X.shape[0]
+        n_cols = X.shape[1]
+
+        labels_dtype = self.classes_.dtype
+
+        if cp.sparse.isspmatrix(X):
+            X = X.tocoo()
+
+            count_features_coo = count_features_coo_kernel(X.dtype,
+                                                           labels_dtype)
+            count_features_coo((math.ceil(X.nnz / 32),), (32,),
+                               (counts,
+                                X.row,
+                                X.col,
+                                X.data,
+                                X.nnz,
+                                n_rows,
+                                n_cols,
+                                Y,
+                                self.n_classes_, False))
+
+        else:
+
+            count_features_dense = count_features_dense_kernel(X.dtype,
+                                                               labels_dtype)
+            count_features_dense((math.ceil(n_rows / 32),
+                                  math.ceil(n_cols / 32), 1),
+                                 (32, 32, 1),
+                                 (counts,
+                                  X,
+                                  n_rows,
+                                  n_cols,
+                                  Y,
+                                  self.n_classes_,
+                                  False,
+                                  X.flags["C_CONTIGUOUS"]))
+
+        count_classes = count_classes_kernel(X.dtype, labels_dtype)
+        count_classes((math.ceil(n_rows / 32),), (32,),
+                      (class_c, n_rows, Y))
+
+        self.feature_count_ += counts
+        self.class_count_ += class_c
+
+    def _update_class_log_prior(self, class_prior=None):
+
+        if class_prior is not None:
+
+            if class_prior.shape[0] != self.n_classes:
+                raise ValueError("Number of classes must match "
+                                 "number of priors")
+
+            self.class_log_prior_ = cp.log(class_prior)
```

Comment:
rmm_cupy_ary

Reply:
Same as above, I believe this should be inheriting the allocator from `class_prior`